api/public/public_api/endpoints/wake_word_file.py

```diff
@@ -0,0 +1,169 @@
+# Mycroft Server - Backend
+# Copyright (C) 2020 Mycroft AI Inc
+# SPDX-License-Identifier: 	AGPL-3.0-or-later
+#
+# This file is part of the Mycroft Server.
+#
+# The Mycroft Server is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+"""Public Device API endpoint for uploading a sample wake word for tagging."""
+from datetime import datetime
+from http import HTTPStatus
+from logging import getLogger
+from os import environ
+from pathlib import Path
+
+from flask import jsonify
+from schematics import Model
+from schematics.types import StringType
+from schematics.exceptions import DataError
+
+from selene.api import PublicEndpoint
+from selene.data.account import Account, AccountRepository
+from selene.data.tagging import (
+    TaggingFileLocationRepository,
+    WakeWordFile,
+    WakeWordFileRepository,
+)
+from selene.data.wake_word import WakeWordRepository
+
+LOCAL_IP = "127.0.0.1"
+
+_log = getLogger(__package__)
+
+
+class UploadRequest(Model):
+    """Data class for validating the content of the POST request."""
+
+    wake_word = StringType(required=True)
+    engine = StringType(required=True)
+    timestamp = StringType(required=True)
+    model = StringType(required=True)
+
+
+class WakeWordFileUpload(PublicEndpoint):
+    """Endpoint for submitting and retrieving wake word sample files.
```

Comment:
I thought wake_word_endpoint.py was the get entry point?

Reply:
This endpoint is in the public device API, which is how devices communicate with Selene.  Devices will never request wake word samples from Selene so all that is needed here is a POST request.  The GET request will be defined in a new Precise API that will be defined when the tagger is implemented.